### ISSUE TITLE
Notebook page-reordered signal handler - conversion failure

### ISSUE THREAD

**Issue by pekim** [diamondburned/gotk4#61]
```go
notebook := gtk.NewNotebook()
notebook.ConnectPageReordered(func(child gtk.Widgetter, pageNum uint) {
	fmt.Println("page re-ordered")
})
```

When the `page-reordered` signal is triggered, a panic results because of a failure to convert an argument.

```
panic: reflect.Value.Convert: value of type *gtk.Widget cannot be converted to type uint [recovered]
	panic: closure error: unexpected panic caught: reflect.Value.Convert: value of type *gtk.Widget cannot be converted to type uint
```

**Comment by diamondburned** [comment]
This is kind of weird. The `Connect` implementation currently "tries" to marshal the instance parameter (`*gtk.Notebook` in this case) to see if the closure needs the 0th argument or not. In this case, `gtk.Widgetter` happens to satisfy the 0th argument check, so the 1st argument (child) is mistakenly unmarshaled into `pageNum`.

I think this should be fixed in the code generator; I'll probably add a `ConnectCapture` method just for the code generator to use that doesn't have the 0th argument check.

**Comment by diamondburned** [comment]
As a workaround, you can do

```go
notebook.Connect("page-reordered", func(notebook *gtk.Notebook, child gtk.Widgetter, pageNum uint) {
	fmt.Println("page re-ordered")
})
```

**Comment by diamondburned** [comment]
This should be fixed in 9ee6939dde9a8f0cf17dd24e91bfde1b0dbc12de.

Can you recheck the `ConnectPageReordered` code?

**Comment by pekim** [comment]
Yes this is now fixed. Thank you.
